spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -2517,4 +2517,16 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
       checkSparkAnswer(df.select("arrUnsupportedArgs"))
     }
   }
+  test("array_remove") {
+    Seq(true, false).foreach { dictionaryEnabled =>
+      withTempDir { dir =>
+        val path = new Path(dir.toURI.toString, "test.parquet")
+        makeParquetFileAllTypes(path, dictionaryEnabled, 10000)
+        spark.read.parquet(path.toString).createOrReplaceTempView("t1");
+        checkSparkAnswerAndOperator(sql("SELECT array_remove(array(_2, _3,_4), _2) from t1"))
+        checkSparkAnswerAndOperator(sql("SELECT array_remove(array(_2, _3,_4), 3) from t1"))
+        checkSparkAnswerAndOperator(sql("SELECT array_remove(array(_2, _3,_4), 10) from t1"))
```

Comment:
Can we add test case where key is null ?

Reply:
Also can we tests where the array is null, array is empty, and array contains nulls.